src/patch.jl

```diff
@@ -45,6 +45,46 @@ end
 
 PatchEnv(debug::Bool=false) = PatchEnv(Dict{Any,Vector{Function}}(), debug)
 
+function patches(pe::PatchEnv)
+    #! format: off
+    return [
+        Patch(target, func)
+        for (target, alternate_funcs) in pe.mapping
+        for func in alternate_funcs
+    ]
+    #! format: on
+end
+
+function Base.:(==)(pe1::PatchEnv, pe2::PatchEnv)
+    return pe1.mapping == pe2.mapping && pe1.debug == pe2.debug
+end
+
+"""
+    merge(pe1::PatchEnv, pe2::PatchEnv)
+
+Merge the two `PatchEnv` instances.
+
+This is done in such a way that the following always holds:
+
+```
+patches_1 = Patch[...]
+patches_2 = Patch[...]
+patches = vcat(patches_1, patches_2)
+
+pe1 = PatchEnv(patches_1)
+pe2 = PatchEnv(patches_2)
+pe = PatchEnv(patches)
+
+@assert pe == merge(pe1, pe2)
+```
+
+The `debug` flag will be set to true if either `pe1` or `pe2` have it set to true.
+"""
+function Base.merge(pe1::PatchEnv, pe2::PatchEnv)
+    new_patches = vcat(patches(pe1), patches(pe2))
+    return PatchEnv(new_patches, pe1.debug || pe2.debug)
```

Comment:
Should avoid having to create intermediate `Patch` instances and eliminates the need for the `patches` function
```suggestion
    mapping = mergewith(vcat, pe1.mapping, pe2.mapping)
    return PatchEnv(mapping, pe1.debug || pe2.debug)
```

Reply:
Ah, this is what I tried originally; the problem is that we can't call that constructor... it ends up dispatching to one that expects an iterable of `Patch` objects.

Reply:
Huh, apparently I was mistaken — in any case, it seems to work now! Changing...